vm/devices/user_driver/src/emulated.rs

```diff
@@ -263,6 +263,26 @@ impl HostDmaAllocator for EmulatedDmaAllocator {
         memory.as_slice().atomic_fill(0);
         Ok(memory)
     }
+
+    fn attach_dma_buffer(&self, len: usize, _base_pfn: u64) -> anyhow::Result<MemoryBlock> {
+        // For emulated allocator (unit tests only) reuse the regular alloc.
+        self.allocate_dma_buffer(len)
```

Comment:
Ditto here, you cannot ignore the restore part.

Ultimately, we should rebase EmulatedDmaAllocator on Chris's page allocator. @chris-oo, can we do this later, as part of separating out the HCL-specific code from the page allocator?

Reply:
Since this is emulated device for unit tests, I am not sure if I can do a real restore here. If I remove this then I also have to remove the corresponding unit test which will start failing.

Reply:
Is the proposal to fail this, then do a follow up to have the `EmulatedDmaAllocator` use the page_allocator? I think that's reasonable to me.

Reply:
Done